systemd/systemd.go

```diff
@@ -1065,18 +1082,53 @@ func (l Log) PID() string {
 	return "-"
 }
 
+type UnitLifetime int
+
+const (
+	Persistent UnitLifetime = iota
+	Transient
+)
+
 // MountUnitPath returns the path of a {,auto}mount unit
 func MountUnitPath(baseDir string) string {
 	escapedPath := EscapeUnitNamePath(baseDir)
 	return filepath.Join(dirs.SnapServicesDir, escapedPath+".mount")
 }
 
+// MountUnitPathWithLifetime returns the path of a {,auto}mount unit
+// created in the systemd directory suitable for the given unit lifetime
+func MountUnitPathWithLifetime(lifetime UnitLifetime, baseDir string) string {
+	escapedPath := EscapeUnitNamePath(baseDir)
+	var servicesPath string
+	switch lifetime {
+	case Persistent:
+		servicesPath = dirs.SnapServicesDir
+	case Transient:
+		servicesPath = dirs.SnapRuntimeServicesDir
+	default:
+		panic("unknown systemd unit lifetime")
+	}
+	return filepath.Join(servicesPath, escapedPath+".mount")
+}
+
+// ExistingMountUnitPath finds the location of an existing mount unit
+func ExistingMountUnitPath(baseDir string) string {
```

Comment:
same comments as for the previous helper

Reply:
Done